File: app/src/main/java/org/michaelbel/moviemade/ui/ReviewActivity.java
package org.michaelbel.moviemade.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.widget.LinearLayout;

import org.michaelbel.moviemade.R;
import org.michaelbel.moviemade.model.MovieRealm;
import org.michaelbel.moviemade.mvp.base.BaseActivity;
import org.michaelbel.moviemade.rest.model.Movie;
import org.michaelbel.moviemade.rest.model.v3.Review;
import org.michaelbel.moviemade.ui.fragment.ReviewFragment;
import org.michaelbel.moviemade.utils.AndroidUtilsDev;

public class ReviewActivity extends BaseActivity {

    public Toolbar toolbar;
    public LinearLayout authorLayout;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_review);

        toolbar = findViewById(R.id.toolbar);
        toolbar.setLayoutParams(AndroidUtilsDev.getLayoutParams(toolbar));
        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(view -> finish());

        authorLayout = findViewById(R.id.author_layout);

        Review review = getIntent().getParcelableExtra("review");
        Movie movie = (Movie) getIntent().getSerializableExtra("movie");
        MovieRealm movieRealm = getIntent().getParcelableExtra("movieRealm");

        if (movieRealm != null) {
            startFragment(ReviewFragment.newInstance(review, movieRealm), R.id.fragment_view);
        } else {
            startFragment(ReviewFragment.newInstance(review, movie), R.id.fragment_view);
        }
    }
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/features/reviews/activity/ReviewActivity.kt
package org.michaelbel.moviemade.presentation.features.reviews.activity

import android.os.Bundle
import kotlinx.android.synthetic.main.activity_default.*
import org.michaelbel.moviemade.R
import org.michaelbel.moviemade.core.entity.Movie
import org.michaelbel.moviemade.core.entity.Review
import org.michaelbel.moviemade.core.local.Intents.EXTRA_MOVIE
import org.michaelbel.moviemade.core.local.Intents.EXTRA_REVIEW
import org.michaelbel.moviemade.core.text.SpannableUtil
import org.michaelbel.moviemade.presentation.base.BaseActivity
import org.michaelbel.moviemade.presentation.features.reviews.fragment.ReviewFragment

class ReviewActivity: BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_default)

        val movie = intent.getSerializableExtra(EXTRA_MOVIE) as Movie
        val review = intent.getSerializableExtra(EXTRA_REVIEW) as Review

        setSupportActionBar(toolbar)
        toolbar.setNavigationOnClickListener { finish() }
        supportActionBar?.title = movie.title
        supportActionBar?.subtitle = SpannableUtil.boldText(getString(R.string.review_by), getString(R.string.review_by, review.author))

        if (savedInstanceState == null) {
            supportFragmentManager
                    .beginTransaction()
                    .replace(container.id, ReviewFragment.newInstance(review))
                    .commit()
        }
    }
}